naga/src/front/wgsl/mod.rs

```diff
@@ -44,6 +44,18 @@ impl Frontend {
     }
 }
 
+/// <div class="warning">
+// NOTE: Keep this in sync with `wgpu::Device::create_shader_module`!
+// NOTE: Keep this in sync with `wgpu_core::Global::device_create_shader_module`!
+///
+/// Note that this function may overflow its stack if the level of recursion in shader parsing
+/// exceeds remaining stack space, resulting in an abortion of the process invoking this function.
+///
+/// Compiler-enforced limits for parsing recursion exist, and if shader compilation runs into
+/// them, it will return an error gracefully. However, you may overflow the stack first if you
+/// do not allocate enough stack space for the thread you invoke this function in!
```

Comment:
todo: Note what users can do. I think just stating, "Make sure you don't overflow the stack with your current shader workload. If you allow users to provide shaders, use enough memory that you can demonstrate running into compiler-enforced limits."

Reply:
Face-to-face feedback from @jimblandy: Please limit yourself to a smaller word budget here; we want to stay within reasonable cognitive resource spend here.

Reply:
Reworded into 6c7ea07ba.